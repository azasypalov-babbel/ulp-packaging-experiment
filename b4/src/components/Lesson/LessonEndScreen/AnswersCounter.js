import React from 'react';
import PropTypes from 'prop-types';
import cx from '../../../lib/cxHelper';
import Counter from './Counter';

const AnswersCounter = ({
  correctItemsCount,
  totalItemsCount,
  onCounterEnd,
  correctAnswersText
}) => {
  return (
    <div
      className={cx('lesson-end-screen-feedback-answers-counter')}
      data-lesson-score={`${correctItemsCount}/${totalItemsCount}`}
    >
      <span className={cx('lesson-end-screen-feedback-answers-counter__caption')}>
        {correctAnswersText}
      </span>
      <h2 className={cx('lesson-end-screen-feedback-answers-counter__counter')}>
        <Counter value={correctItemsCount} />
        <span>/</span>
        <Counter value={totalItemsCount} onCounterEnd={onCounterEnd} />
      </h2>
    </div>
  );
};

AnswersCounter.propTypes = {
  correctItemsCount: PropTypes.number.isRequired,
  totalItemsCount: PropTypes.number.isRequired,
  onCounterEnd: PropTypes.func,
  correctAnswersText: PropTypes.string.isRequired
};

AnswersCounter.defaultProps = {
  onCounterEnd: () => {}
};

export default AnswersCounter;
