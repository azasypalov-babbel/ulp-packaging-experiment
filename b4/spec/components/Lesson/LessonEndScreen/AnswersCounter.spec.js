import React from 'react';
import AnswersCounter from '../../../../src/components/Lesson/LessonEndScreen/AnswersCounter';
import { shallow } from 'enzyme';

describe('Answers Counter', () => {
  test('it renders', () => {
    const props = {
      onCounterEnd: jest.fn(),
      correctAnswersText: 'Answered correctly:',
      correctItemsCount: 8,
      totalItemsCount: 12
    };
    const component = shallow(
      <AnswersCounter {...props} />
    );

    expect(component).toMatchSnapshot();
  });
});
