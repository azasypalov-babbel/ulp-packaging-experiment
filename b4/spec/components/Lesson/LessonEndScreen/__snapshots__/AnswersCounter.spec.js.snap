// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Answers Counter it renders 1`] = `
<div
  className="loy-lesson-end-screen-feedback-answers-counter"
  data-lesson-score="8/12"
>
  <span
    className="loy-lesson-end-screen-feedback-answers-counter__caption"
  >
    Answered correctly:
  </span>
  <h2
    className="loy-lesson-end-screen-feedback-answers-counter__counter"
  >
    <Counter
      onCounterEnd={[Function]}
      value={8}
    />
    <span>
      /
    </span>
    <Counter
      onCounterEnd={[MockFunction]}
      value={12}
    />
  </h2>
</div>
`;
